src/settings.ts

```diff
@@ -13,16 +13,40 @@ export class Settings {
                 return;
             }
             const updatedConfig = workspace.getConfiguration("java.dependency");
-            if (updatedConfig.showOutline !== this._depdendencyConfig.showOutline
-                || updatedConfig.packagePresentation !== this._depdendencyConfig.packagePresentation) {
+            if (updatedConfig.showOutline !== this._dependencyConfig.showOutline
+                || updatedConfig.packagePresentation !== this._dependencyConfig.packagePresentation
+                || (updatedConfig.syncWithFolderExplorer !== this._dependencyConfig.syncWithFolderExplorer
+                    && updatedConfig.syncWithFolderExplorer)) {
                 commands.executeCommand(Commands.VIEW_PACKAGE_REFRESH);
             }
-            this._depdendencyConfig = updatedConfig;
+            this._dependencyConfig = updatedConfig;
 
         }));
 
-        const instrumented = instrumentOperation(Commands.VIEW_PACKAGE_CHANGEREPRESENTATION, Settings.changePackageRepresentation);
-        context.subscriptions.push(commands.registerCommand(Commands.VIEW_PACKAGE_CHANGEREPRESENTATION, instrumented));
+        context.subscriptions.push(commands.registerCommand(Commands.VIEW_PACKAGE_SYNCWITHFOLDER,
+            instrumentOperation(Commands.VIEW_PACKAGE_SYNCWITHFOLDER, Settings.syncWithFolderCommand)));
```

Comment:
Could we use only one command for the operation, just dynamically change the icon?